File: DocViewerDemos/VB/WinForms/DocDocumentViewer/Export/ImageViewer.vb
Imports System.ComponentModel
Imports System.Text

Namespace Export
	Partial Public Class ImageViewer
		Inherits Form
		Public Sub New()
			InitializeComponent()

			Dim bitMapCount As Integer = Form1.bitMap.Count
			Dim pageNum As Integer = Form1.m_CurrentPageNum

			If bitMapCount > 0 Then
				For Each dic As KeyValuePair(Of String, Image) In Form1.bitMap
					Dim filename As String = dic.Key
					Dim bmap As Image = dic.Value
					Dim iv As New ImageViewerControl()
					iv.Image = bmap
					iv.FileName = filename
					iv.Text = pageNum.ToString()
					iv.Cursor = Cursors.Hand
					AddHandler iv.MouseClick, AddressOf iv_MouseClick
					Me.flowLayoutPanel1.Controls.Add(iv)
					iv.Left = CInt(Me.flowLayoutPanel1.Width - iv.Width) / 2
					pageNum += 1
				Next dic
			End If
		End Sub

		Private Sub iv_MouseClick(ByVal sender As Object, ByVal e As MouseEventArgs)
			Dim filename As String = (CType(sender, ImageViewerControl)).FileName
			Dim process As New Process()
			process.StartInfo = New ProcessStartInfo(filename)
			process.Start()
		End Sub
	End Class
End Namespace

File: DocViewerDemos/CS/WinForms/DocDocumentViewer/Export/ImageViewer.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Text;
using System.Windows.Forms;

namespace Export
{
    public partial class ImageViewer : Form
    {
        public ImageViewer()
        {
            InitializeComponent();

            int bitMapCount = Form1.bitMap.Count;
            int pageNum = Form1.m_CurrentPageNum; ;
            if (bitMapCount > 0)
            {
                foreach (KeyValuePair<string, Image> dic in Form1.bitMap)
                {
                    string filename = dic.Key;
                    Image bmap = dic.Value;
                    ImageViewerControl iv = new ImageViewerControl();
                    iv.Image = bmap;
                    iv.FileName = filename;
                    iv.Text = pageNum.ToString();
                    iv.Cursor = Cursors.Hand;
                    iv.MouseClick += new MouseEventHandler(iv_MouseClick);
                    this.flowLayoutPanel1.Controls.Add(iv);
                    iv.Left = (int)(this.flowLayoutPanel1.Width - iv.Width) / 2;
                    pageNum++;
                }
            }
        }

        private void iv_MouseClick(object sender, MouseEventArgs e)
        {
            string filename = ((ImageViewerControl)(sender)).FileName;
            System.Diagnostics.Process process = new System.Diagnostics.Process();
            process.StartInfo = new System.Diagnostics.ProcessStartInfo(filename);
            process.Start();
        }      
    }
}
